text/66-self-managed-stack-sets.md

```diff
@@ -0,0 +1,537 @@
+* A StackSet construct for customers to extend
+* A new assembly/artifact type for the Cloud Assembly
+* A new stack set synthesizer
+* Deployment code that creates/updates stack sets using AWS CloudFormation APIs.
+
+### Is this a breaking change?
+
+No.
+
+### What are the drawbacks of this solution?
+
+The handling of CDK assets is more complex than a normal stack CDK app. 
+Stack set instances are deployed independently from CDK context, and exist in different accounts/regions. 
+This is dicussed more in the design details.
+
+### What alternative solutions did you consider?
+
+An alternative is the usage of `cdk.CfnStackSet`, which is not ideal for customers with existing stack sets since there is significant migration effort.
+
+### What is the high level implementation plan?
+* Create StackSet core construct
+* Create StackSet artifact type in the Cloud Assembly
+* Create Bootstrapless StackSet Synthesizer
+* Update CLI to collect StackSet artifact types in addition to Stacks
+* Deployment code for creating/updating stack sets
+
+### Are there any open issues that need to be addressed later?
+
+* CDK asset handling
+* Future support for service managed stack sets
+* CDK Pipelines
+
+---
+# Implemenation Details
+
+# Requirements
+
+## In Scope Features
+
+* Create Self-Managed Stack Set
+  * This should create the stack set directly, as opposed to the [CfnStackSet](https://docs.aws.amazon.com/cdk/api/latest/docs/@aws-cdk_aws-cloudformation.CfnStackSet.html) construct which is defined in a single CloudFormation stack that deploys the stack set
+* Update Self-Managed Stack Set
+  * Update includes updating **all existing** stack set instances 
+  * Set Operation Preference Configuration
+
+## Out of Scope Features
+
+* Delete Self-Managed Stack Set
+* Service Managed Stack Set
+* Add/Remove stack set instances through CDK
+* Update to only specific stack set instances
+* Stack Set Instance Level Parameter Overrides
+* CDK Assets (CDK File Assets/CDK Docker Assets)
+  * More information on CDK asset handling and complexities described in the [CDK Assets Handling](#4-cdk-assets-handling) section.
+
+## Assumptions
+
+* Parent account contains Stack Set administration role and child accounts contains Stack Set execution role for enabling stack set operations
+  * https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/stacksets-prereqs-self-managed.html
+  * The Stack Set administration role in the parent account is used by CloudFormation to assume the execution role in the child account. This role can be created manually or may be created through a CDK stack. This `role arn` is passed into the stack set CDK construct properties 
+  * The Stack Set execution role in the child is the role assumed by the administration role in the parent account, and utilized to provision resources in the child account. This role should be provisioned manually, and the `role name` is passed into the stack set CDK construct properties
```

Comment:
I assume this was supposed to read "child account".

Reply:
Again, same as above, this is a gigantic bombshell that is not mentioned anywhere in the customer-facing docs of this RFC.

All of this information needs to be moved all the way up to the "working backwards" section of this RFC.

Reply:
Can you explain why wouldn't we add this Role to `cdk bootstrap`, and require a child account be `cdk bootstrap`ped before a StackSet instance of a StackSet managed by the CDK can be created inside of it?

That seems better than requiring a manually-provisioned Role, no?

Reply:
Yeah, my thought process for the stack set administration role in the parent account was it is a property of the CloudFormation stack set vs a property of CDK. That being said, I think customer could provision the stack set administration role through CDK as a single stack.

As for the stack set execution role, this one is kind of odd in a sense that it is managed outside of the CDK context. If a customer defines all the child accounts in their CDK app, then they could provision the execution role from CDK, but question might be what is the value of stack sets.

Reply:
Talked offline, will include more details and bootstrap workflow for parent and child accounts